src/socket/address.cr

```diff
@@ -219,7 +231,23 @@ class Socket
     def to_unsafe : LibC::Sockaddr*
       sockaddr = Pointer(LibC::SockaddrUn).malloc
       sockaddr.value.sun_family = family
-      sockaddr.value.sun_path.to_unsafe.copy_from(@path.to_unsafe, @path.bytesize + 1)
+
+      if @abstract
+        raw_path = '\0' + @path
+      else
+        raw_path = @path
+      end
+      sockaddr.value.sun_path.to_unsafe.copy_from(raw_path.to_unsafe, raw_path.bytesize + 1)
```

Comment:
You're right, I got confused with copy/from source/dest, where is the null byte prepended, ...